apps/prairielearn/src/components/Navbar.html.ts

```diff
@@ -557,7 +557,7 @@ function AuthnOverrides({
       <label class="visually-hidden" for="effective-uid">UID</label>
       <input
         id="effective-uid"
-        type="email"
+        type="text"
```

Comment:
Was this the fix for #11751? If so, can you explain how that fixed the error message you saw there? I can't immediately see the connection.

Reply:
The form had failed input validation, and since the button type is submit, but it was submitted through JS, it threw that error.

Reply:
https://stackoverflow.com/a/28340579

Reply:
I can update the button type too, but this fixes an assumption that the UID is always an email

Reply:
I think the correct fix here is going to be to just add a `name` to the input. In practice, UIDs indeed must be a thing that looks like an email. From the docs you cited:

> This [the UID] should generally be an email-like identifier

Maybe this needs stronger language?

Reply:
I had tried adding a name to the input, that doesn't fix the issue. The issue is that `type="button"` needs to be set for a JS button according to the SO link.

Reply:
I think the problem is actually that the dropdown is hidden when the button is clicked, and since the input then ends up inside a DOM subtree with `display: none`, the input can't be focused. The error message could have made that more clear...

I think the right fix will be to add [`data-bs-auto-close="outside"`](https://getbootstrap.com/docs/5.3/components/dropdowns/#auto-close-behavior) to the dropdown so that it doesn't close when it's interacted with.

Reply:
Added in [1c3adaf](https://github.com/PrairieLearn/PrairieLearn/pull/11752/commits/1c3adafcb4978fdee0aad04cfa47aa238c6d6ca9).